src/components/data/data-utils.js

```diff
@@ -928,6 +928,104 @@ export const MultipleEntityEditor = ({ entityType, entityNames, attributeNames,
+  const [columnName, setColumnName] = useState('')
+  const [columnNameTouched, setColumnNameTouched] = useState(false)
+  const columnNameError = columnNameTouched && Utils.cond(
+    [!columnName, () => 'A column name is required.'],
+    [_.includes(columnName, entityMetadata[entityType].attributeNames), () => 'This column already exists.']
+  )
+
+  const [value, setValue] = useState('')
+
+  const [isBusy, setIsBusy] = useState()
+
+  const addColumn = async () => {
+    try {
+      setIsBusy(true)
+
+      const queryResults = await Ajax().Workspaces.workspace(namespace, name).paginatedEntitiesOfType(entityType, {
+        pageSize: entityMetadata[entityType].count,
+        fields: ''
+      })
+      const allEntityNames = _.map(_.get('name'), queryResults.results)
+
+      const entityUpdates = _.map(entityName => ({
+        entityType,
+        name: entityName,
+        attributes: { [columnName]: prepareAttributeForUpload(value) }
+      }), allEntityNames)
+
+      await Ajax()
+        .Workspaces
+        .workspace(namespace, name)
+        .upsertEntities(entityUpdates)
+      onSuccess()
+    } catch (e) {
+      onDismiss()
+      reportError('Unable to add column.', e)
+    }
+  }
+
+  return h(Modal, {
+    title: 'Add a new column',
+    onDismiss,
+    okButton: h(ButtonPrimary, {
+      disabled: !columnName || columnNameError,
+      tooltip: columnNameError,
+      onClick: addColumn
+    }, ['Save'])
+  }, [
+    div({ style: { display: 'flex', flexDirection: 'column', marginBottom: '1rem' } }, [
+      h(IdContainer, [
+        id => h(Fragment, [
+          label({ htmlFor: id, style: { fontWeight: 'bold', marginBottom: '0.5rem' } }, 'Column name'),
+          div({ style: { position: 'relative', display: 'flex', alignItems: 'center' } }, [
+            h(TextInput, {
+              id,
+              value: columnName,
+              placeholder: 'Enter a name (required)',
+              style: columnNameError ? {
+                paddingRight: '2.25rem',
+                border: `1px solid ${colors.danger()}`
+              } : undefined,
+              onChange: value => {
+                setColumnName(value)
+                setColumnNameTouched(true)
+              }
+            }),
+            columnNameError && icon('error-standard', {
+              size: 24,
+              style: {
+                position: 'absolute', right: '0.5rem',
+                color: colors.danger()
+              }
+            })
+          ]),
+          columnNameError && div({
+            'aria-live': 'assertive',
+            'aria-relevant': 'all',
+            style: {
+              marginTop: '0.5rem',
+              color: colors.danger()
+            }
+          }, columnNameError)
+        ])
+      ])
+    ]),
+    p([
+      span({ style: { fontWeight: 'bold' } }, ['Default value']),
+      ' (will be entered for all rows)'
```

Comment:
Watching the videos you attached to the PR (think you for that!), it wasn't obvious to me that this was optional, probably because the placeholder above that said "required" was no longer rendered. I think I would like it to be more obvious that this field is optional, either by including that in the placeholder text here or using a "*" or other persistent indicator on the required fields.

Reply:
In a way, it's not optional. If nothing is entered, the default value of an empty string will be used.

Reply:
Updated the label in https://github.com/DataBiosphere/terra-ui/pull/3017/commits/28ad6d4e7241f4a790be3916a2f7fbe2130859bc.